eras/alonzo/impl/src/Cardano/Ledger/Alonzo/PlutusScriptApi.hs

```diff
@@ -147,8 +136,10 @@ knownToNotBe1Phase scriptsAvailable (sp, sh) = do
 collectTwoPhaseScriptInputs ::
   forall era.
   ( EraTx era,
-    ShelleyEraTxBody era,
+    ShelleyMAEraTxBody era,
     AlonzoEraTxWits era,
+    EraUTxO era,
+    ScriptsNeeded era ~ [(ScriptPurpose (EraCrypto era), ScriptHash (EraCrypto era))],
```

Comment:
Would it be clearer, and more consistent with the rest of the code, if we used a new type for `[(ScriptPurpose (EraCrypto era), ScriptHash (EraCrypto era))]` and called it something like `AlonzoScriptsNeeded`?

Reply:
ha ha, I should have read the whole PR before commenting. It seem you've already done exactly this elsewhere. So maybe just this spot and a couple others need the newtype instead?

Reply:
Done.